labs/b9.md

```diff
@@ -0,0 +1,178 @@
+
+Here is an example `Dockerfile` that will build an image based on Fedora Linux that has `python3.6` installed by default. Before building the `Dockerfile`, make sure to first run `docker pull fedora` to retrieve the image from Docker Hub. The code for this exercise can be found in `~/decal-labs/b9/missile`.    
+
+```docker  
+FROM fedora:latest  
+
+RUN yum update -y && yum install python3-pip && \     
+    pip3 install --upgrade pip && \     
+    pip3 install pyfiglet termcolor    
+
+ADD missile.py /  
+CMD ["/usr/bin/python3.6", "missile.py"]    
+```
+
+Note: there are some "best practices" for writing Dockerfiles that the above example doesn’t use, because it’s a basic example. If you’re interested in this stuff, [check out this article.](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/)    
+
+What is this doing? We first specify a base image `fedora:latest` (Fedora 27). We then specify that we should run (`RUN`) the command `yum update` (`yum` is Fedora's package manager), and then `yum install pip3` installs `pip` for `python3`. Then we install the `pyfiglet` and `termcolor` packages via `pip3` since our python script depends on them. We add the script for our machine to the container, and lastly we run the script with `python3.6`.   Take a moment to appreciate how cool this is. We have a completely different Linux distribution running on our system that ships with a version of python that is not in `apt` on your student VMs!
+
+Build the `Dockerfile` with the following:
+
+`docker build -t missile:latest .`    
+
+This tells Docker to look in the current directory for a `Dockerfile` to build, and builds it. The `-t` flag tells Docker to tag this build with the name `missile:latest`. Note that building the missile image will take a couple of minutes to complete.      
+
+You can see all of the images you’ve built on your machine with the `docker images` command.    
+
+##### Questions    
+
+1. Run the image you just built. As before, don’t forget the `-it` flags, and you can exit using CTRL+D. What do you observe?    
+2. Write and build a `Dockerfile` based on `ubuntu:xenial` that installs the packages `fortune` and `fortunes-min` and runs the fortune executable (located in `/usr/games/fortune` after you install it). Note that you won’t need to use the `-it` flags when you run the container as fortune doesn’t need `STDIN`. Submit your `Dockerfile` with this lab.  Hint: if you’re having trouble writing your Dockerfile, try booting an interactive container and installing both packages. How can you translate what you did interactively to a Dockerfile?   
+3. Paste the output running `docker images` command after questions 1 and 2.      
+
+#### Dockerizing a Web Application    
```

Comment:
I don't think you'd normally consider apache a web application, so I'd change all references to this

Reply:
Yeah, whoops. This was left was last semester.